tools/Azure.Mcp.Tools.Kusto/tests/Azure.Mcp.Tools.Kusto.LiveTests/KustoCommandTests.cs

```diff
@@ -84,6 +90,83 @@ public async Task Should_list_kusto_tables()
         Assert.NotEmpty(tablesArray.EnumerateArray());
     }
 
+    [Fact]
+    public async Task Should_list_tables_with_direct_uri()
+    {
+        var clusterInfo = await CallToolAsync(
+            "azmcp_kusto_cluster_get",
+            new()
+            {
+                { "subscription", Settings.SubscriptionId },
+                { "cluster", Settings.ResourceBaseName }
+            });
+
+        var clusterUri = clusterInfo.AssertProperty("cluster").AssertProperty("clusterUri").GetString();
+        Assert.NotNull(clusterUri);
+
+        var result = await CallToolAsync(
+            "azmcp_kusto_table_list",
+            new()
+            {
+                { "cluster-uri", clusterUri },
+                { "database", TestDatabaseName }
+            });
+
+        var tablesArray = result.AssertProperty("tables");
+        Assert.Equal(JsonValueKind.Array, tablesArray.ValueKind);
+        Assert.NotEmpty(tablesArray.EnumerateArray());
+    }
+
+    [Fact]
+    public async Task Should_get_table_schema()
+    {
+        var result = await CallToolAsync(
+            "azmcp_kusto_table_schema",
+            new()
+            {
+                { "subscription", Settings.SubscriptionId },
+                { "cluster", Settings.ResourceBaseName },
+                { "database", TestDatabaseName },
+                { "table", TestTableName }
+            });
+
+        var schema = result.AssertProperty("schema").GetString();
+        Assert.NotNull(schema);
+        Assert.Contains("Title:string", schema);
+        Assert.Contains("IsCompleted:bool", schema);
+    }
+
+    [Fact]
+    public async Task Should_get_table_schema_with_direct_uri()
+    {
+        var clusterInfo = await CallToolAsync(
+            "azmcp_kusto_cluster_get",
+            new()
+            {
+                { "subscription", Settings.SubscriptionId },
+                { "cluster", Settings.ResourceBaseName }
+            });
+
+        var clusterUri = clusterInfo.AssertProperty("cluster").AssertProperty("clusterUri").GetString();
```

Comment:
this assumes you have access to the resource. this won't be true if we intend to use something like the help cluster, so worth adding a comment about that. in any case, good for now.